examples/eventProvider.js

```diff
@@ -6,15 +6,44 @@
 * EVENT_PROVIDER is smart enough to filter broadcasted events.
 */
 
+/*
+  to build this sample run
+
+    npm run examples:build
+
+  to run
+
+    npm run examples:run:eventprovider
+
+  if this fails, ensure you have to latest version of spark-protocol, run
+
+    npm update spark-protocol
+
+*/
+
 /* eslint-disable */
-import type { Event } from '../src/types';
 
+import type { Event } from '../types';
 import { Container } from 'constitute';
-import { defaultBindings } from 'spark-server';
+import defaultBindings from '../defaultBindings';
+import settings from '../settings.js';
+import logger from '../lib/logger';
 
 const container = new Container();
 defaultBindings(container, settings);
 
-container.constitute('EVENT_PROVIDER').onNewEvent((event: Event) => {
-  // do piping stuff here.
+const deviceServer = container.constitute('DeviceServer');
+deviceServer.start();
+
+/*
+const evProvider = container.constitute('EVENT_PROVIDER');
+evProvider.onNewEvent((event: Event) => {
+  logger.info('Event onNewEvent',event);
+});
+*/
+
+
+
+deviceServer._eventPublisher.subscribe('*', (ev: any) => {
```

Comment:
So, after the correction in spark-protocol, you don't need to use eventPublisher directly, just eventProvider.